wagtail_newsletter/campaign_backends/mailchimp.py

```diff
@@ -61,3 +111,100 @@ def get_audience_segments(self, audience_id) -> "list[AudienceSegment]":
             )
             for segment in segments
         ]
+
+    def _create_campaign(self, body) -> str:
+        body = copy(body)
+        body.setdefault("type", "regular")
+        try:
+            return cast(str, self.client.campaigns.create(body)["id"])
+
+        except ApiClientError as error:
+            log_and_raise(error, "Error while creating campaign")
+
+    def _update_campaign(self, campaign_id: str, body) -> str:
+        try:
+            self.client.campaigns.update(campaign_id, body)
+            return campaign_id
+
+        except ApiClientError as error:
+            if error.status_code == 404:
+                # The campaign was deleted on the Mailchimp end; create a new one
+                return self._create_campaign(body)
+
+            log_and_raise(
+                error, "Error while updating campaign", campaign_id=campaign_id
+            )
+
+    def _set_content(self, campaign_id: str, html: str):
```

Comment:
I find `-> None` hints useful, as they immediately signal that a function/method is purely used for its "side effects", without returning anything

Reply:
That makes sense! I've added it. Hopefully I didn't miss any other functions.